device_specific/flash_the_ttgo_tdisplay

```diff
@@ -81,25 +93,183 @@ fi
 cd "${jade_save_directory}"
 jade_version="$(git describe --tags)"
 
+pip install cbor2
+export PYTHONPATH="$jade_save_directory"
+
+secure_boot_status=$(
+  python << EOF
+from jadepy import JadeAPI
+import _cbor2
+import serial
+
+create_jade_fn = JadeAPI.create_serial
+kwargs = {'device': None, 'timeout': 5}
+
+try:
+    with create_jade_fn(**kwargs) as jade:
+        verinfo = jade.get_version_info()
+
+    if 'SB' in verinfo['JADE_FEATURES']:
+        print('enabled')
+    else:
+        print('disabled')
+except _cbor2.CBORDecodeEOF:
+    print('uninitialized')
+except serial.serialutil.SerialException:
+    print('devicenotfound')
+EOF
+)
+
+clear
+if [ "$secure_boot_status" = "enabled" ]; then
+  echo "Secure-boot is enabled."
+  enable_secure_boot=u
+elif [ "$secure_boot_status" = "disabled" ]; then
+  echo "Secure-boot is disabled."
+  echo ""
+  echo "Would you like to enable Secure Boot?"
+  echo "Your options are:"
+  echo "  y - Yes"
+  echo "  n - No"
+  echo ""
+  echo "Warning: Enabling secure boot is only recommended after making sure that everything works. DO NOT ENABLE IF YOU ARE NOT SURE ABOUT THE CONSEQUENCES"
+  if [ "${CI:-false}" = true ]; then
+    enable_secure_boot=n
+  else
+    while true; do
+      read -r -p "Please select an option (y/n): " enable_secure_boot
+      case $enable_secure_boot in
+      [Yy]*)
+        echo "Secure Boot is being enabled..."
+        break
+        ;;
+      [Nn]*)
+        echo "Secure Boot will not be enabled."
+        break
+        ;;
+      *) echo "Invalid option, please enter y or n" ;;
+      esac
+    done
+  fi
+elif [ "$secure_boot_status" = "uninitialized" ]; then
+  echo "The device is not initialized yet."
+  echo ""
+  echo "Would you like to enable Secure Boot?"
+  echo "Your options are:"
+  echo "  y - Yes"
+  echo "  n - No"
+  echo ""
+  echo "Warning: Enabling secure boot is only recommended after making sure that everything works. DO NOT ENABLE IF YOU ARE NOT SURE ABOUT THE CONSEQUENCES"
+  if [ "${CI:-false}" = true ]; then
+    enable_secure_boot=n
+  else
+    while true; do
+      read -r -p "Please select an option (y/n): " enable_secure_boot
+      case $enable_secure_boot in
+      [Yy]*)
+        echo "Secure Boot is being enabled..."
+        break
+        ;;
+      [Nn]*)
+        echo "Secure Boot will not be enabled."
+        break
+        ;;
+      *) echo "Invalid option, please enter y or n" ;;
+      esac
+    done
+  fi
+else
+  echo "secure_boot_status is $secure_boot_status"
+  [ "${CI:-false}" = false ] && echo "This should only happen within ci builds, check manually" && exit 1
+  enable_secure_boot=n
+fi
+
+[ -f sdkconfig ] && mv sdkconfig sdkconfig.bak # make sure sdkconfig is removed from previous builds
+
 cp configs/sdkconfig_display_ttgo_tdisplay.defaults sdkconfig.defaults
-sed -i.bak '/CONFIG_DEBUG_MODE/d' ./sdkconfig.defaults
```

Comment:
Just curious - was this causing errors or a style preference?

Reply:
Just a style preference to shorten the code